### DISCUSSION TITLE
Issue with ProcessOut transitive dependency: cmark

### DISCUSSION THREAD

**Discussion by dcacenabes** [swiftyfinch/Rugby#382] [Q&A] [answered]
In our project we use a dependency that is not compatible with Rugby. The library is [ProcessOut](https://github.com/processout/processout-ios/blob/efb4dc4eca4ed04df8d706f3be8020bba66b8aed/ProcessOut.podspec), and the transitive dependency is a precompiled xcframework: [cmark](https://github.com/processout/processout-ios/blob/efb4dc4eca4ed04df8d706f3be8020bba66b8aed/ProcessOut.podspec#L12).

I iterated over the sample project I shared recently in other issue, simply adding this dependency to the Podfile.
[RugbyIssueDemoUseFrameworks.zip](https://github.com/swiftyfinch/Rugby/files/14726055/RugbyIssueDemoUseFrameworks.zip)

### Steps to reproduce:
- `bundle exec pod install`
- `rugby --config DebugStaging`
- Build the app target

### Desired output:
App is built successfully

### Actual output:
App build fails with error:
Framework 'cmark' not found
Linker command failed with exit code 1

![Screenshot 2024-03-22 at 17 55 58](https://github.com/swiftyfinch/Rugby/assets/3796899/971b5a62-4fb4-4ef3-9920-c20c759d64bb)

### Current workaround:
Excluding ProcessOut works perfectly, but of course we lose some build speed.
`rugby -x "ProcessOut" --config DebugStaging`

**Comment by swiftyfinch** [comment, marked as answer]
Hello @dcacenabes!

I'm working on fixing this soon.
The problem occurs when using the "XCFramework" script phase in combination with the "sources" phase.
We need to separate the "XCFramework" script phase from the built target to create a separate aggregate target.

And again thanks a lot for your detailed description and an example 👍
It's helped me a lot.

**Reply by swiftyfinch** [reply]
@dcacenabes Could you test this one #384?

You can check out this branch and then run `make install`.
This command will build and replace your current 🏈 Rugby version.

**Reply by dcacenabes** [reply]
Sure thing! Right now 👍

**Reply by dcacenabes** [reply]
Tested it in our prod project and works as expected. No more issue with cmark. Thanks!

**Reply by swiftyfinch** [reply]
🚢 [v2.8.0](https://github.com/swiftyfinch/Rugby/releases/tag/2.8.0)
